.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,44 @@
+# build from source of datree
+- id: datree-test-golang-built
+  name: datree test, build datree and run it (golang is dependency)
+  description: Prevent Kubernetes misconfigurations from reaching production
+  entry: datree
+  args: ['test', '--only-k8s-files']
+  types: [file, yaml]
+  language: golang
+  pass_filenames: true
+
+# use a pre-installed binary of datree, user must run install.sh prior to using pre-commit hook
+- id: datree-test-pre-installed
+  name: datree test, datree is already installed locally
+# Via the current install.sh? his reduces the effectiveness of pre-commit
+# now as our CI tool package manager - no version pinning in .pre-commit-config.yaml :(
+  description: Prevent Kubernetes misconfigurations from reaching production
+  entry: /usr/local/bin/datree
+  args: ['test', '--only-k8s-files']
+  types: [file, yaml]
+  language: system
+
+# use Docker to run datree, use :latest or semantic versioning tag in dockerhub
+# to pin a particular release
+- id: datree-test-docker
+  name: datree test, run within docker
+  description: Prevent Kubernetes misconfigurations from reaching production
+  types: [file, yaml]
+  language: docker_image
+  pass_filenames: true
+  args: ['test', '--only-k8s-files']
+
+  # entry: datree/datree:latest
+  # this errors out :(
+  # starting container process caused: exec: "./datree": stat ./datree: no such file or directory: unknown.
+
+  # entry: --entrypoint './datree test --only-k8s-files' datree/datree:latest
+  # this errors out:
+  # starting container process caused: exec: "./datree test --only-k8s-files": stat ./datree test --only-k8s-files: no such file or directory: unknown.
+
+  # I updated your Dockerfile. Includes update to go 1.17 and it now works when I point to a local repo built with
+  # the same Dockerfile. I also added a multi-stage build which brought container size down from ~430mb to ~20mb
+  entry: datree-test:latest
```

Comment:
1. change entry to the datree image from our repo in dockerhub, when we will merge this pr it will work.
2. comments will be redundant at this point so you can remove it, thanks for the clarification btw :)

Reply:
@MentalBrake 

1. I will change this back to `datree/datree:latest`

In regards to best practices for `rev` key: Unfortunately it looks like the `docker_image` type doesn't support the `rev` option. Their documentation dictates that the `entry` key contains the tag/version we'd like to use. For users that just want the latest available `datree/datree:latest` is ideal. If a user wants to specify a version there, they'd choose something like `datree/datree:v1.42.0`.